components/tabs/tab.js

```diff
@@ -0,0 +1,132 @@
+import '../colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
```

Comment:
`<d2l-tab>` seems to be a "private" element -- like it's only used by `<d2l-tabs>` right? Wondering if we could do more to "hide" it so that others don't mistakenly use it....

Reply:
At one point I'd gone down the path of baking this into `d2l-tabs` completely, but it tends to bloat the code.  I could move the component into the `d2l-tabs.js` file so that it's not quite so "public"?  It would avoid the extra HTTP request too.  That still would not prevent someone from using it though. There might be a way to create the elements inside of `render` using the Tab class, without having to use `customElements.define('d2l-tab', Tab);`... if so, that would truly make it private.

Reply:
Possibly yeah. Or you could register it as `d2l-tab-internal-private-do-not-use-extra-rich-and-unsafe` haha.

Reply:
I renamed `d2l-tab` to `d2l-tab-internal`.  While not strictly private, it should make it clear that consumers should not try to use it.

I tried constructing the element from `Tab` and not using the `customElements.define` API but I didn't have much luck.  While it's simple enough to construct `Tab`, I wasn't able to figure out how to append it to the DOM without the define API... there might be a way, but it's certainly not "advertised" and probably not recommended.

Reply:
Makes sense -- I think what you have should suffice!